test/parallel/test-fs-promises-watch.js

```diff
@@ -123,6 +123,27 @@ assert.rejects(
   },
   { code: 'ERR_INVALID_ARG_TYPE' }).then(common.mustCall());
 
+assert.rejects(
+  async () => {
+    // eslint-disable-next-line no-unused-vars, no-empty
+    for await (const _ of watch('', { maxQueue: 'silly' })) { }
+  },
+  { code: 'ERR_INVALID_ARG_TYPE' }).then(common.mustCall());
+
+assert.rejects(
+  async () => {
+    // eslint-disable-next-line no-unused-vars, no-empty
+    for await (const _ of watch('', { overflow: 1 })) { }
+  },
+  { code: 'ERR_INVALID_ARG_VALUE' }).then(common.mustCall());
+
+assert.rejects(
+  async () => {
+    // eslint-disable-next-line no-unused-vars, no-empty
+    for await (const _ of watch('', { overflow: 'barf' })) { }
+  },
+  { code: 'ERR_INVALID_ARG_VALUE' }).then(common.mustCall());
+
```

Comment:
The previous tests are actually not ideal to follow because like all the other async fs APIs, `fs.watch()` throws validation errors synchounously so there's no need to use `assert.rejects`.

```suggestion
assert.throws(() => watch('', { maxQueue: 'silly' }), { code: 'ERR_INVALID_ARG_TYPE' });
assert.throws(() => watch('', { overflow: 1 }), { code: 'ERR_INVALID_ARG_TYPE' });
assert.throws(() => watch('', { overflow: 'barf' }), { code: 'ERR_INVALID_ARG_TYPE' });
```

Reply:
I have to revert these as they don't throw synchronously since `fs.watch` is an `async` function.